File: Data_Structure/python/7785.py
n = int(input())
name_list = set();
for _ in range(n):
  name_,order = input().split()
  if order == "enter":
    name_list.add(name_)
  else:
    name_list.remove(name_)
print("\n".join(sorted(name_list,reverse=True)))

File: Data_Structure/javascript/7785.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");

const answer = [
  ...input.slice(1, -1).reduce((answer, row) => {
    const [name_, order] = row.split(" ");
    if (order === "enter") {
      answer.add(name_);
    } else {
      answer.delete(name_);
    }
    return answer;
  }, new Set()),
]
  .sort()
  .reverse()
  .join("\n");

console.log(answer);
